extensions/ql-vscode/src/databases/db-config-store.ts

```diff
@@ -2,23 +2,31 @@ import * as fs from 'fs-extra';
 import * as path from 'path';
 import { cloneDbConfig, DbConfig } from './db-config';
 import * as chokidar from 'chokidar';
+import { DisposableObject } from '../pure/disposable-object';
 
-export class DbConfigStore {
+export class DbConfigStore extends DisposableObject {
   private readonly configPath: string;
 
   private config: DbConfig;
+  private configWatcher: chokidar.FSWatcher | undefined;
 
   public constructor(workspaceStoragePath: string) {
+    super();
     this.configPath = path.join(workspaceStoragePath, 'dbconfig.json');
 
     this.config = this.createEmptyConfig();
+    this.configWatcher = undefined;
   }
 
   public async initialize(): Promise<void> {
     await this.loadConfig();
     this.watchConfig();
   }
 
+  public dispose(): void {
+    this.configWatcher?.unwatch(this.configPath);
+  }
```

Comment:
Is it worth adding a little test to check we call dispose correctly?

There are some examples in our [telemetry tests](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/src/vscode-tests/no-workspace/telemetry.test.ts#L133).

Reply:
Hmm that's an interesting idea! 

I think there are two types of tests we could potentially add:
- One that tests that the `dispose` function is actually clearing out the watcher. I think that would require some `proxyquire` stuff 🤔 and adding more usage of `proxyquire` feels like adding tech debt since we want to move away from it soon. 
- One further up the chain (so can't write it yet) that uses `DbConfigStore` and ensures that `dispose` is called on it at the right time (like the example you linked).

So I think I'll just note this in an issue somewhere for now and come back to it later on. If we've not moved away from `proxyquire` by then, we might just have to use it.

Thanks for the feedback!